sw/host/tests/chip/sysrst_ctrl/sysrst_ctrl_outputs.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{ensure, Result};
+use clap::Parser;
+use once_cell::sync::Lazy;
+use std::collections::HashMap;
+use std::fs;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use object::{Object, ObjectSymbol};
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::io::uart::Uart;
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::test_utils::mem::MemWriteReq;
+use opentitanlib::test_utils::test_status::TestStatus;
+use opentitanlib::uart::console::UartConsole;
+use opentitanlib::{collection, execute_test};
+
+use sysrst_ctrl::{Config, setup_pins, set_pins, read_pins};
+
+#[derive(Debug, Parser)]
+struct Opts {
+    #[command(flatten)]
+    init: InitializeTest,
+
+    /// Console receive timeout.
+    #[arg(long, value_parser = humantime::parse_duration, default_value = "10s")]
+    timeout: Duration,
+
+    /// Path to the firmware's ELF file, for querying symbol addresses.
+    #[arg(value_name = "FIRMWARE_ELF")]
+    firmware_elf: PathBuf,
+}
+
+struct Params<'a> {
+    opts: &'a Opts,
+    transport: &'a TransportWrapper,
+    uart: &'a dyn Uart,
+    config: &'a Config,
+    test_phase_addr: u32,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[repr(u8)]
+enum TestPhase {
+  // There is a Setup phase but this is the initial value on the device code so we do not need it.
+    Loopback = 1,
+    OverrideSetup = 2,
+    OverrideZeros = 3,
+    OverrideOnes = 4,
+    OverrideRelease = 5,
+    OverrideAndLoopback = 6,
+    Done = 7,
+}
+
+static CONFIG: Lazy<HashMap<&'static str, Config>> = Lazy::new(|| {
+    collection! {
+        "hyper310" => Config {
+            // key0_in, key1_in, key2_in, pwrb_in
+            output_pins: vec!["IOR10", "IOR11", "IOR12", "IOR5", ],
+            open_drain: vec![false, false, false, false],
+            // key0_out, key1_out, key2_out, pwrb_out, bat_dis_out, z3_wakeup_out, ec_rst, flash_wp
```

Comment:
I wonder if we should use aliases for these pins so we can refer to them directly by name. Don't worry about it for this PR, though. I know the alias-mapping code can get quite spaghetti.

Reply:
Except for ec_rst and flash_wp, those MIOS that can be routed to any pin though so I am not sure an alias is ideal. But I think applying your suggestion to those two pins makes sense though! I'll create a separate PR for that.